// 0utmailcore/index.js
// index.js
const API_URL = "https://api0utmail-test-email.vercel.app";

// Internal helper
const post = async (endpoint, payload) => {
    try {
        const res = await fetch(`${API_URL}${endpoint}`, {
            method: "POST",
            headers: { "Content-Type": "application/json" },
            body: JSON.stringify(payload)
        });
        return await res.json();
    } catch (err) {
        return { success: false, error: err.message };
    }
};

/**
 * Send a plain text email.
 * @param {string} apiKey - Your Firebase User Key
 * @param {object} token - Your Google Token Object
 * @param {string} to - Recipient Email
 * @param {string} subject - Email Subject
 * @param {string} text - Email Body
 */
exports.send = (apiKey, token, to, subject, text) =>
    post("/send", { api_key: apiKey, google_token: token, to, subject, text });

/**
 * Send an HTML email.
 * @param {string} apiKey - Your Firebase User Key
 * @param {object} token - Your Google Token Object
 * @param {string} to - Recipient Email
 * @param {string} subject - Email Subject
 * @param {string} html - HTML Body
 */
exports.sendHtml = (apiKey, token, to, subject, html) =>
    post("/sendHtml", { api_key: apiKey, google_token: token, to, subject, html });

/**
 * Read the inbox.
 * @param {string} apiKey - Your Firebase User Key
 * @param {object} token - Your Google Token Object
 * @param {number} limit - Number of emails (default 10)
 */
exports.read = (apiKey, token, limit = 10) =>
    post("/read", { api_key: apiKey, google_token: token, limit });
